### PR TITLE
[sovereign] Sovereign requesters

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- Sovereign requesters
  
## Proposed changes
- Extra fixes for requesting trie nodes
- Apply a light version of body to header func for epoch change

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by axenteoctavian — APPROVED** [review]

File: process/block/sovereignChainBlock.go [hunk 1]
```diff
@@ -1918,8 +1961,6 @@ func (scbp *sovereignChainBlockProcessor) updateState(header data.HeaderHandler,
 				log.Warn("couldn't commit trie checkpoint", "epoch", sovHdr.GetEpoch(), "error", err)
 			}
 		}()
-
-		scbp.nodesCoordinator.ShuffleOutForEpoch(header.GetEpoch())
```
Comment (sasurobert) [inline process/block/sovereignChainBlock.go:1922]: why is this deleted ?

File: dataRetriever/requestHandlers/requestHandler.go [hunk 2]
```diff
@@ -502,12 +532,8 @@ func (rrh *resolverRequestHandler) RequestTrieNode(requestHash []byte, topic str
 		"chunk index", chunkIndex,
 	)
 
-	requester, err := rrh.requestersFinder.MetaChainRequester(topic)
+	requester, err := getRequesterFunc(topic)
```
Comment (sasurobert) [inline dataRetriever/requestHandlers/requestHandler.go:535]: can you make a better name for getRequesterFunc ?

File: process/block/sovereignChainBlock.go (hunk 1 above)
Comment (mariusmihaic) [reply process/block/sovereignChainBlock.go:1922]: It was wrongfully put there in the beginning.
if we respect the metaBlock processor, this should only happen in `commitEpochStart`, which also happens for sovereign

File: dataRetriever/requestHandlers/requestHandler.go (hunk 2 above)
Comment (mariusmihaic) [reply dataRetriever/requestHandlers/requestHandler.go]: Right, renamed them

File: process/block/sovereignChainBlock.go (hunk 1 above)
Comment (mariusmihaic) [reply process/block/sovereignChainBlock.go:1922]: So it should have never been put in `updateState`

**Review by sasurobert — APPROVED** [review]

**Review by axenteoctavian — APPROVED** [review]
